thirdparty/cnpy/cnpy.cpp

```diff
@@ -356,10 +360,15 @@ cnpy::NpyArray cnpy::npy_load(std::string fname) {
 
     if(!fp) throw std::runtime_error("npy_load: Unable to open file "+fname);
 
-    NpyArray arr = load_the_npy_file(fp);
+    try {
+        NpyArray arr = load_the_npy_file(fp);
+        fclose(fp);
+        return arr;
+    }catch (const std::exception & ex) {
+        fclose(fp);
+        throw ex;
+    }
```

Comment:
```suggestion
    }catch (...) {
        fclose(fp);
        throw;
    }
```

`throw ex` will slice off the original type of the exception, reducing it to just `std::exception`.